go/engine/kex2_provisioner.go

```diff
@@ -344,3 +381,34 @@ func (e *Kex2Provisioner) rememberDeviceInfo(jw *jsonw.Wrapper) error {
 
 	return nil
 }
+
+func (e *Kex2Provisioner) makeSdhBoxes(receiverKeyGeneric libkb.GenericKey) (res []keybase1.SharedDhSecretKeyBox, err error) {
+	receiverKey, ok := receiverKeyGeneric.(libkb.NaclDHKeyPair)
+	if !ok {
+		return res, fmt.Errorf("Unexpected receiver key type")
+	}
+
+	sdhk, err := e.G().GetSharedDHKeyring()
+	if err != nil {
+		return res, err
+	}
+	err = sdhk.Sync(e.ctx.NetContext)
+	if err != nil {
+		return res, err
+	}
```

Comment:
You probably want to bail out here if there is no shared DH key for this user, don't you? We don't want to be introducing a shared DH on Kex2 provisioning, that sounds wrong.